dot/network/host.go

```diff
@@ -26,13 +28,36 @@ import (
 	ma "github.com/multiformats/go-multiaddr"
 )
 
-var privateCIDRs = []string{
-	"10.0.0.0/8",
-	"172.16.0.0/12",
-	"192.168.0.0/16",
-	"100.64.0.0/10",
-	"198.18.0.0/15",
-	"169.254.0.0/16",
+func newPrivateIPFilters() (privateIPs *ma.Filters, err error) {
+	privateCIDRs := []string{
+		"10.0.0.0/8",
+		"172.16.0.0/12",
+		"192.168.0.0/16",
+		"100.64.0.0/10",
+		"198.18.0.0/15",
+		"169.254.0.0/16",
+	}
+	privateIPs = ma.NewFilters()
```

Comment:
nit use that instead

```go
func IPIsPrivate(ip net.IP) bool {
	return ip.IsPrivate() || ip.IsLoopback() ||
		ip.IsLinkLocalUnicast() || ip.IsLinkLocalMulticast()
}
```

Reply:
Also note the current implementation doesn't work with IPv6 or IPv4-in-IPv6 addresses.